pkg/cluster/sync.go

```diff
@@ -403,7 +403,10 @@ func (c *Cluster) syncStatefulSet() error {
 		c.logger.Warnf("could not get list of pods to apply PostgreSQL parameters only to be set via Patroni API: %v", err)
 	}
 
-	requiredPgParameters := c.Spec.Parameters
+	requiredPgParameters := make(map[string]string)
+	for k, v := range c.Spec.Parameters {
+		requiredPgParameters[k] = v
+	}
```

Comment:
Probably, `maps.Clone` can be used instead as 
`requiredPgParameters := maps.Clone(c.Spec.Parameters)`

Reply:
Oh, `maps.Copy` should be used in this case only